### ISSUE TITLE
Feature request: async version of map, andThen

### ISSUE THREAD

**Issue by Jack-Works** [vultix/ts-results#33]
```ts
result.andThen(async e => calc(e))
```

**Comment by andreitopli** [comment]
@shalexbas I agree with you. We have been using [ts-async-result](https://github.com/movesthatmatter/ts-async-results) in production extensively on [https://www.chessroulette.live](https://www.chessroulette.live) and it works perfectly, haven't encountered any issues yet. I think there's a thread here opened about merging _ts-async-results_ into this library.

**Comment by GabrielCTroia** [comment]
Or someone could help me maintain it 😁! But yeah, we’ve been using it extensively at chessroulette so we will hopefully be able to put more resources into it as we grow.

**Comment by ctsstc** [comment]
I was looking around for this, but I'm sad to find this open ticket that such does not exist.

For now I'll continue with this pattern:

```typescript
async function anAsyncMethod(): Promise<Result<SomeType, SomeError>> {
  const result = await someAsyncCall()

  if (result.ok) {
    return await someOtherAsyncCall()
  }

  return result
}

async function someAsyncCall(): Promise<Result<SomeType, SomeError>> {}

async function someOtherAsyncCall(): Promise<Result<SomeType, SomeError>> {}
```

There's also scenarios where you may have to map the types as well.
